src/ray/util/logging.cc

```diff
@@ -96,22 +96,19 @@ class SpdLogMessage final {
   explicit SpdLogMessage(const char *file, int line, int loglevel) : loglevel_(loglevel) {
     stream() << ConstBasename(file) << ":" << line << ": ";
   }
-  inline std::shared_ptr<spdlog::logger> get_logger() {
-    auto logger = spdlog::get("ray_log_sink");
-    if (!logger) {
-      logger = spdlog::get("stderr");
-    }
+  inline std::shared_ptr<spdlog::logger> GetDefaultLogger() {
     // We just emit all log informations to stderr when no default logger has been created
     // before starting ray log, which is for glog compatible.
-    if (!logger) {
-      logger = spdlog::stderr_color_mt("stderr");
-      logger->set_pattern(RayLog::GetLogFormatPattern());
-    }
+    static auto logger = spdlog::stderr_color_mt("stderr");
+    logger->set_pattern(RayLog::GetLogFormatPattern());
```

Comment:
Somebody may say `set_pattern` could not be thread-safe. To simplify to fix this issue, this pr will gurantee that get stderr logger is safe, but I found set_pattern is low cost and thread-safe made in spdlog source code 
```cpp
// Set global formatter. Each sink in each logger will get a clone of this object
SPDLOG_INLINE void registry::set_formatter(std::unique_ptr<formatter> formatter)
{
    std::lock_guard<std::mutex> lock(logger_map_mutex_);
    formatter_ = std::move(formatter);
    for (auto &l : loggers_)
    {
        l.second->set_formatter(formatter_->clone());
    }
}


```